src/commands/createVirtualMachine/VirtualMachineCreateStep.ts

```diff
@@ -70,6 +70,11 @@ export class VirtualMachineCreateStep extends AzureWizardExecuteStep<IVirtualMac
         context.virtualMachine = await computeClient.virtualMachines.createOrUpdate(rgName, vmName, virtualMachineProps);
         ext.outputChannel.appendLog(createdVm);
 
+        context.telemetry.properties.os = context.os;
```

Comment:
Move this stuff to the very beginning of the step so that we will have it if an error occurs during the step